discovery/azure/azure.go

```diff
@@ -372,6 +376,48 @@ func (d *Discovery) refresh(ctx context.Context) ([]*targetgroup.Group, error) {
 	return []*targetgroup.Group{&tg}, nil
 }
 
+// isRunningVM return true if the VM is up and running
+// ref: https://docs.microsoft.com/en-us/azure/virtual-machines/windows/states-lifecycle
+func (client *azureClient) isRunningVM(ctx context.Context, vm *compute.VirtualMachine) bool {
+	r, err := newAzureResourceFromID(*vm.ID, client.logger)
+	if err != nil {
+		return false
+	}
+	instanceView, err := client.vm.InstanceView(ctx, r.ResourceGroup, *vm.Name)
+	if err != nil {
+		level.Error(client.logger).Log("msg", "failed to get InstanceView", "VM name", *vm.Name, "err", err)
+		return false
+	}
+	for _, status := range *instanceView.Statuses {
+		if strings.HasPrefix(*status.Code, powerStatePrefix) && strings.Compare(*status.Code, powerStateRunning) != 0 {
```

Comment:
check for `status.Code == nil`. I'm also not sure why we would need to check both for the prefix and the full string: `if status.Code != nil && *status.Code == powerStateRunning {` should be enough.

Reply:
For EC2 we just expose the state, without filtering it as depending on use case there you might want to notice that an instance is stopped.